daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -2146,15 +2146,13 @@ case class FNTwoArgsMathExpr(nameAsParsed: String, fnQName: RefQName,
 case class DFDLXShiftExpr(nameAsParsed: String, fnQName: RefQName,
                              args: List[Expression], constructor: (List[CompiledDPath], NodeInfo.Kind) => RecipeOp)
   extends FunctionCallBase(nameAsParsed, fnQName, args) {
-  checkArgCount(2)
   override lazy val inherentType = {
-
     val argInherentType = args(0).inherentType
     schemaDefinitionUnless(
       argInherentType.isSubtypeOf(NodeInfo.PrimType.UnsignedLong) ||
         argInherentType.isSubtypeOf(NodeInfo.PrimType.Long),
       "First argument must be either xs:unsignedLong or xs:long or a subtype of those, but was %s.",
```

Comment:
I'm thinking we can improve this diagnostic if we include the function name? So if we change the error message to:

> First argument for %s must be either ...

And pass in ``nameAsParsed``

Then we get an error message like "First argument for dfdlx:leftShift must be either xs:unsigned ...". Makes it a little more clear and lets you check for xs:left/rightShift in the tdml:error's.